Nunez deploy docs 3 (#13949)
PrefectHQ/prefect: docs/3.0rc/deploy/dynamic-infra/deploy-docker.mdx

- Deploy and run your Docker image on a Kubernetes cluster.

> - Look at the Prefect-maintained Docker images and discuss options for use.

Reviewer: The Docker image selection strategy section also appeals (and will mostly be used by folks who are using work-pool based deployments. Should we break it out into its own page in the Resources section. cc: @billpalombi

    ```

>     After running the above command, the container should start up and serve the flow within the container.

Reviewer: ```suggestion
    After running the above command, the container should start and the flow should run within the container.
```

  </Tab>
</Tabs>

## Deploy to a remote environment

Now that you have a Docker image with your flow code embedded, you can deploy it to a remote environment.

This guide simulates a remote environment by using Kubernetes locally with Docker Desktop. 
Use the instructions provided by Docker to 
[set up Kubernetes locally.](https://docs.docker.com/desktop/kubernetes/)

### Create a Kubernetes deployment manifest

To ensure the process serving your flow is always running, 
create a [Kubernetes deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/). 

> If your flow's container ever crashes, Kubernetes automatically restarts it, 

Reviewer: ```suggestion
If your flow's container crashes, Kubernetes automatically restarts it, 
```

            # Never pull the image because we're using a local image
            imagePullPolicy: Never
    ```
<Tip>
**Keep your API key secret**

> In the above manifest, you are passing in the Prefect API URL and API key as environment variables. 

Reviewer: ```suggestion
In the above manifest, you are passing the Prefect API URL and API key as environment variables. 
```

            # Never pull the image because we're using a local image
            imagePullPolicy: Never
    ```
<Tip>
**Linux users**

If you're running Linux, set your `PREFECT_API_URL` to use the IP address of your 
machine instead of `host.docker.internal`.
</Tip>
  </Tab>
</Tabs>

This manifest defines how your image runs when deployed in your Kubernetes cluster. 
Note that you will run a single replica of your flow container. 
If you want to run multiple replicas of your flow container to keep up with an active schedule, 
or because your flow is resource-intensive, you can increase the `replicas` value.

### Deploy your flow to the cluster

> You now have a deployment manifest. To deploy your flow to the cluster, run:

Reviewer 2: ```suggestion
You now have a deployment manifest. To deploy your flow to the cluster, use:
```

```bash
kubectl apply -f deployment-manifest.yaml
```

> To monitor the status of your Kubernetes deployment, run:

Reviewer 2: ```suggestion
To monitor the status of your Kubernetes deployment, use:
```

```bash
kubectl get deployments
```

Once the deployment successfully starts, check the logs of your flow container 
by running the following:

```bash
kubectl logs -l flow=get-repo-info
```

> You're now serving your flow in your cluster. To trigger a flow, run:

Reviewer 2: ```suggestion
You're now serving your flow in your cluster. To trigger a flow run, use:
```

> For example, when release `2.11.5` is published:

Reviewer: ```suggestion
For example, when release `3.1.1` is published:
```

1. Images with the release packaged are built for each supported Python version 

> (3.8, 3.9, 3.10, 3.11) with both standard Python and Conda.

Reviewer: ```suggestion
(3.9, 3.10, 3.11, 3.12) with both standard Python and Conda.
```
